lib/spack/spack/util/typing.py

```diff
@@ -27,3 +37,7 @@ def __gt__(self, other: Any) -> bool:
 
     def __ge__(self, other: Any) -> bool:
         raise NotImplementedError
+
+
+if not TYPE_CHECKING:
+    SupportRichComparison = object
```

Comment:
I think there's a typo:
```suggestion
    SupportsRichComparison = object
```